diff --git a/net/miniupnpd/Makefile b/net/miniupnpd/Makefile
index 64e7667a4..cfba85c50 100644
--- a/net/miniupnpd/Makefile
+++ b/net/miniupnpd/Makefile
@@ -26,7 +26,7 @@ PKG_BUILD_PARALLEL:=1
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/version.mk
 
-define Package/miniupnpd
+define Package/miniupnpd/Default
   SECTION:=net
   CATEGORY:=Network
   DEPENDS:=+iptables +libip4tc +IPV6:libip6tc +IPV6:ip6tables +libuuid +libcap-ng
@@ -35,10 +35,26 @@ define Package/miniupnpd
   URL:=https://miniupnp.tuxfamily.org/
 endef
 
+define Package/miniupnpd
+  $(call Package/miniupnpd/Default)
+  TITLE+= (IGDv2)
+  CONFLICTS:=miniupnpd-igdv1
+  VARIANT:=igdv2
+endef
+
+define Package/miniupnpd-igdv1
+  $(call Package/miniupnpd/Default)
+  TITLE+= (IGDv1)
+  PROVIDES:=miniupnpd
+  VARIANT:=igdv1
+endef
+
 define Package/miniupnpd/conffiles
 /etc/config/upnpd
 endef
 
+Package/miniupnpd-igdv1/conffiles = $(Package/miniupnpd/conffiles)
+
 define Build/Prepare
 	$(call Build/Prepare/Default)
 	echo "$(VERSION_NUMBER)" | tr '() ' '_' >$(PKG_BUILD_DIR)/os.openwrt
@@ -46,12 +62,15 @@ endef
 
 CONFIGURE_ARGS = \
 	$(if $(CONFIG_IPV6),--ipv6) \
-	--igd2 \
 	--leasefile \
 	--portinuse \
 	--firewall=iptables \
 	--disable-fork
 
+ifeq ($(BUILD_VARIANT),igdv2)
+	CONFIGURE_ARGS += --igd2
+endif
+
 TARGET_CFLAGS += $(FPIC) -flto
 TARGET_LDFLAGS += -Wl,--gc-sections,--as-needed
 
@@ -71,4 +90,7 @@ define Package/miniupnpd/install
 	$(INSTALL_DATA) ./files/firewall.include $(1)/usr/share/miniupnpd/firewall.include
 endef
 
+Package/miniupnpd-igdv1/install = $(Package/miniupnpd/install)
+
 $(eval $(call BuildPackage,miniupnpd))
+$(eval $(call BuildPackage,miniupnpd-igdv1))
\ No newline at end of file
